BaragonService/src/main/java/com/hubspot/baragon/service/worker/BaragonStatePersisterWorker.java

```diff
@@ -0,0 +1,35 @@
+package com.hubspot.baragon.service.worker;
+
+import java.util.Collection;
+import java.util.ServiceLoader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hubspot.baragon.data.BaragonStateDatastore;
+import com.hubspot.baragon.models.BaragonServiceState;
+import com.hubspot.baragon.state.persister.BaragonStatePersister;
+
+public class BaragonStatePersisterWorker implements Runnable {
+  private static final Logger LOG = LoggerFactory.getLogger(BaragonStatePersisterWorker.class);
+
+  private final ServiceLoader<BaragonStatePersister> loader;
+  private final BaragonStateDatastore stateDatastore;
+
+  private BaragonStatePersisterWorker(BaragonStateDatastore stateDatastore) {
+    this.loader = ServiceLoader.load(BaragonStatePersister.class);
+    this.stateDatastore = stateDatastore;
+  }
+
+  @Override
+  public void run() {
+    Collection<BaragonServiceState> state = stateDatastore.getGlobalState();
+    int version = stateDatastore.getStateVersion().or(-1);
```

Comment:
Hm, what's the semantic meaning of sending a `stateVersion` of `-1` here? Would that potentially break things? If there's no state version, should we bail out instead of filling in a default?

Reply:
It was handled because we later check if the new version is greater than the previous version so things won't break, but changed it if the optional has a value to remove the extra work